internal/services/containers/kubernetes_cluster_node_pool_resource_test.go

```diff
@@ -547,20 +547,42 @@ func TestAccKubernetesClusterNodePool_spot(t *testing.T) {
 	})
 }
 
-func TestAccKubernetesClusterNodePool_upgradeSettings(t *testing.T) {
+func TestAccKubernetesClusterNodePool_upgradeSettingsMaxSurge(t *testing.T) {
 	data := acceptance.BuildTestData(t, "azurerm_kubernetes_cluster_node_pool", "test")
 	r := KubernetesClusterNodePoolResource{}
 
 	data.ResourceTest(t, r, []acceptance.TestStep{
 		{
-			Config: r.upgradeSettings(data, 35, 18),
+			Config: r.upgradeSettingsMaxSurge(data, `10%`, 10, 5, "Cordon"),
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
 			),
 		},
 		data.ImportStep(),
 		{
-			Config: r.upgradeSettings(data, 5, 0),
+			Config: r.upgradeSettingsMaxSurge(data, `33%`, 15, 8, "Schedule"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccKubernetesClusterNodePool_upgradeSettingsMaxUnavailable(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_kubernetes_cluster_node_pool", "test")
+	r := KubernetesClusterNodePoolResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.upgradeSettingsMaxUnavailable(data, `25%`, 10, 5),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.upgradeSettingsMaxUnavailable(data, `50%`, 15, 8),
```

Comment:
could we add more testing to ensure that `max_surge` and `max_unavailable` can be removed and that the config can be updated to switch between using `max_surge` and `max_unavailable`?